cmd/serverless/dependency_list_test.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test && linux
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const erroMsg = `
+The dependencies.txt file is out of date. Please run: CGO_ENABLED=1 GOOS=linux go list -f '{{join .Deps "\n"}}' -tags serverless github.com/DataDog/datadog-agent/cmd/serverless > cmd/serverless/dependencies.txt to update it
+`
+
+func buildDependencyList() (string, error) {
+	run := "go"
```

Comment:
I think you can have an easier command using `-deps` instead of the custom formatting ?

Reply:
The `-deps` also display dependencies of dependencies.

> The -deps flag causes list to iterate over not just the named packages
but also all their dependencies. It visits them in a depth-first post-order
traversal, so that a package is listed only after all its dependencies.

https://pkg.go.dev/cmd/go/internal/list

We just want to the top level dependencies 😄. I think using `go list -f` works good for our use case